sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_transport_async.py

```diff
@@ -483,11 +482,8 @@ async def _read(self, n, buffer=None, **kwargs): # pylint: disable=unused-argume
 
     def close(self):
         """Do any preliminary work in shutting down the connection."""
-        # TODO: async close doesn't:
-        # 1) shutdown socket and close. --> self.sock.shutdown(socket.SHUT_RDWR) and self.sock.close()
```

Comment:
checking that self.ws.shutdown(socket.SHUT_RDWR) doesn't need to be called before the ws.close() call below?

Reply:
yep, the `websocket-client` library does exactly that when you can call wsl.close() ... [link](https://github.com/websocket-client/websocket-client/blob/61171591b08ee031e02cc6cb129952259062f502/websocket/_core.py#L497) to library code